src/timesync/timesyncd-bus.c

```diff
@@ -23,24 +42,86 @@ static int property_get_servers(
                 void *userdata,
                 sd_bus_error *error) {
 
-        ServerName *p, **s = userdata;
-        int r;
+        ServerName **s = userdata;
 
         assert(s);
         assert(bus);
         assert(reply);
 
-        r = sd_bus_message_open_container(reply, 'a', "s");
+        return reply_server_names(*s, reply);
+}
+
+static int property_get_runtime_servers(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Manager *m = userdata;
+
+        assert(m);
+
+        return reply_server_names(m->runtime_servers, reply);
+}
+
+static int property_set_runtime_servers(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *message,
+                void *userdata,
+                sd_bus_error *error) {
+
+        _cleanup_strv_free_ char **names = NULL;
+        Manager *m = userdata;
+        char **name;
+        int r;
+
+        assert(m);
+        assert(bus);
+        assert(message);
+
+        r = sd_bus_message_read_strv(message, &names);
         if (r < 0)
                 return r;
 
-        LIST_FOREACH(names, p, *s) {
-                r = sd_bus_message_append(reply, "s", p->string);
+        while (m->runtime_servers)
+                server_name_free(m->runtime_servers);
+
+        STRV_FOREACH(name, names) {
+                r = dns_name_is_valid_or_address(*name);
```

Comment:
please validate all params first. and only after that is complete aallocate ServerName objects for them. i.e. we should strive to implement all-or-nothing behaviour here.